### DISCUSSION TITLE
Is it possible to join tables on a custom field?

### DISCUSSION THREAD

**Discussion by mshidlov** [prisma/prisma#17659] [Q&A]
Is this possible to do a join in Prisma query using custom join?

I have a use case when I'm trying to query 2 connection tables, and I want to skip the main table join.

For example, I have a class with students and teachers.

```
model Class {
  id             BigInt          @id @default(autoincrement())
  name           String          
  student_class  student_class[]
  teacher_class  teacher_class[]
}

model student_class {
  class_id       BigInt
  student_id     BigInt
  student        student    @relation(fields: [student_id], references: [id])
  Class          Class      @relation(fields: [class_id], references: [id])

  @@id([class_id, student_id])
  @@index([student_id])
  @@index([class_id])
}

model teacher_class {
  class_id       BigInt
  teacher_id     BigInt
  teacher        teacher    @relation(fields: [student_id], references: [id])
  Class          Class      @relation(fields: [class_id], references: [id])

  @@id([class_id, student_id])
  @@index([teacher_id])
  @@index([class_id])
}
```
 
If I want to get a teacher, students I will have to do a query that will join through the class table:
```
this.prisma.student_class.findMany({
            where:{
                Class:{
                    teacher_class:{
                        some:{
                            teacher_id: teacher_id
                        }
                    }
                }
            }
        })
````

What I like to do is a direct join between student_class and teacher_class:

```sql
SELECT * 
FROM student_class sc
JOIN teacher_class tc ON sc.class_id= tc.class_id
WHERE tc.id = $teacher_id
```

**Comment by nurul3101** [comment]
Hello @mshidlov 👋 

Based on the schema file, I don't think it's possible as there is no direct relation between the `student_class` and `teacher_class` models. It would be necessary to go through the intermediary model of `Class` that you are currently doing.
The other alternative would be to directly use [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access).

**Reply by mshidlov** [reply]
Hey @nurul3101,
Thank you for your answer 👍 

Those are the options that I know.

The first one will produce a redundant Join operation that I'm trying to avoid, and the second one is doing raw SQL, and for that, we don't need Prisma.

If this is not possible today, it could be a good enhancement in the future.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
